### PR TITLE
Enable Type2 modification and refactor Info.Marshal

### PR DESCRIPTION
**Pull request by stevenh-tw**

### CODE REVIEW

File: pkg/smbios/info64_test.go [hunk 1]
```diff
@@ -116,33 +133,48 @@ func Test64Marshal(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Error setup mock system info table: %v", err)
 	}
-	longTable.strings[0] += "a-very-long-string"
+	longStr := "a-very-loooooooooooooooooooooooooooooooooooong-string"
```
Comment (hugelgupf) [inline pkg/smbios/info64_test.go:136]: do you think we could convert this test to one that doesn't use the mock data?

I think it'd be clearer what's being tested and how if we went from []*Table -> modify -> binary data, or binary data -> modify -> binary data in a tabled test.

File: pkg/smbios/smbios_modifier.go [hunk 2]
```diff
@@ -42,28 +41,96 @@ func getEntries(smbiosBase func() (int64, int64, error), memFile *os.File) (*Ent
...
 	}
 }
 
-// ModifySystemInfo modifies the SystemInfo table in system memory
-func (m *Modifier) ModifySystemInfo(manufacturer, productName, version, serialNumber string) error {
-	sysInfo, err := m.Info.GetSystemInfo()
-	if err != nil {
-		return fmt.Errorf("failed to get system info: %w", err)
+// ReplaceBaseboardInfoMotherboard returns override options that only overrides table with Type = BaseboardInfo and BoardType = BoardTypeMotherboardIncludesProcessorMemoryAndIO
+func ReplaceBaseboardInfoMotherboard(assetTag *string) OverrideOpt {
```
Comment (hugelgupf) [inline pkg/smbios/smbios_modifier.go:99]: I think this is ok for now, but there's gotta be a way to have a better API. I'm good to merge this one though and improve it in the other repo.

File: pkg/smbios/smbios_modifier.go [hunk 3]
```diff
@@ -42,28 +41,96 @@ func getEntries(smbiosBase func() (int64, int64, error), memFile *os.File) (*Ent
...
+			if t.Type != TableTypeBaseboardInfo {
+				result = append(result, t)
+				continue
+			}
+
+			bi, err := ParseBaseboardInfo(t)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse baseboard info: %w", err)
+			}
+			if bi.BoardType != BoardTypeMotherboardIncludesProcessorMemoryAndIO {
```
Comment (hugelgupf) [inline pkg/smbios/smbios_modifier.go:112]: can you cover this case with a test case?

File: pkg/smbios/smbios_modifier.go (hunk 2 above)
Comment (stevenh-tw) [reply pkg/smbios/smbios_modifier.go:99]: Noted, thanks.

File: pkg/smbios/info64_test.go [hunk 4]
```diff
@@ -88,94 +88,124 @@ func Test64GetTablesByType(t *testing.T) {
 	}
 }
 
-func systemInfoTable() (*Table, error) {
-	info, err := setupMockData()
-	if err != nil {
-		return nil, err
+func defulatType1Table() *Table {
```
Comment (rafaelcn) [inline pkg/smbios/info64_test.go:91]: shouldn't it be `defaultType1Table`?

Comment (rafaelcn) [reply pkg/smbios/info64_test.go:91]: the same goes for the other occurrences of the same name

**Review by rafaelcn — COMMENTED** [review]
Did I find some typos or is it my mistake?

File: pkg/smbios/info64_test.go (hunk 4 above)
Comment (stevenh-tw) [reply pkg/smbios/info64_test.go]: Yes, it's a typo, thanks

File: pkg/smbios/type2_baseboard_information.go [hunk 5]
```diff
@@ -28,6 +29,92 @@ type BaseboardInfo struct {
...
+		d = append(d, 0)
+	}
+
+	chb := make([]byte, 2)
+	binary.LittleEndian.PutUint16(chb, bi.ChassisHandle)
+	d = append(d, chb...)
+
+	d = append(d, byte(bi.BoardType))
+	d = append(d, bi.NumberOfContainedObjectHandles)
+
```
Comment (hugelgupf) [inline pkg/smbios/type2_baseboard_information.go:103]: check that NumberOf... == len(bi.Contained...)?

**Pull request merged**